gr-uhd/lib/usrp_source_impl.cc

```diff
@@ -203,6 +203,46 @@ ::uhd::gain_range_t usrp_source_impl::get_gain_range(const std::string& name, si
     return _dev->get_rx_gain_range(name, chan);
 }
 
+bool usrp_source_impl::has_power_reference(const size_t chan)
+{
+#ifdef UHD_USRP_MULTI_USRP_POWER_LEVEL
+    const size_t dev_chan = _stream_args.channels[chan];
+    return _dev->has_rx_power_reference(dev_chan);
+#else
+    return false;
+#endif
+}
+
+void usrp_source_impl::set_power_reference(const double power_dbm, const size_t chan)
+{
+#ifdef UHD_USRP_MULTI_USRP_POWER_LEVEL
+    const size_t dev_chan = _stream_args.channels[chan];
+    _dev->set_rx_power_reference(power_dbm, dev_chan);
+#else
+    throw std::runtime_error("not implemented in this version");
```

Comment:
I mean here ... this file, for the "source" ...

Reply:
Oh! :D

I was thinking "what's he saying, I've clearly changed the source code.... yup, coming right up.